backends/arm/test/tester/test_pipeline.py

```diff
@@ -0,0 +1,361 @@
+# Copyright 2025 Arm Limited and/or its affiliates.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import logging
+from typing import Any, Callable, Generic, List, TypeVar
+
+import torch
+from executorch.backends.arm.test import common
+from executorch.backends.arm.test.tester.arm_tester import ArmTester
+from executorch.exir.backend.compile_spec_schema import CompileSpec
+
+
+logger = logging.getLogger(__name__)
+T = TypeVar("T")
+""" Generic type used for test data in the pipeline. Depends on which type the operator expects."""
+
+
+class BasePipeline(Generic[T]):
```

Comment:
Nit: To help improve readability

```suggestion
class BasePipelineMaker(Generic[T]):
```

if you are not a fan for `BasePipelineFactory` or something along those lines :)